apps/user-portal/src/components/shared/user-avatar.tsx

```diff
@@ -101,9 +148,152 @@ export const UserAvatar: FunctionComponent<UserAvatarProps> = (props: UserAvatar
         setShowPopup(false);
     };
 
-    // Avatar for the authenticated user.
-    if (authState && authState.profileInfo && (authState.profileInfo.profileUrl || authState.profileInfo.userImage)) {
+    /**
+     * This contains the Avatar component
+     */
+    const avatar = (
+        <Avatar
+            avatarType="user"
+            bordered={ false }
+            image={
+                (authState && authState.profileInfo
+                    && (authState.profileInfo.profileUrl || authState.profileInfo.userImage))
+                    ?
+                    authState.profileInfo.profileUrl
+                        ? authState.profileInfo.profileUrl
+                        : authState.profileInfo.userImage
+                    :
+                    userImage
+            }
+            label={ showGravatarLabel ? resolveTopLabel() : null }
+            name={ authState ? resolveUserDisplayName(authState) : name || "" }
+            onMouseOver={ handleOnMouseOver }
+            onMouseOut={ handleOnMouseOut }
+            { ...rest }
+        />
+    );
+
+    /**
+     * This function services the API call to update the profileUrl
+     */
+    const updateProfileUrl = () => {
+        const data = {
+            Operations: [
+                {
+                    op: "replace",
+                    value: {
+                        profileUrl: url
+                    }
+                }
+            ],
+            schemas: ["urn:ietf:params:scim:api:messages:2.0:PatchOp"]
+        };
+        updateProfileInfo(data).then((response) => {
+            if (response.status === 200) {
+                onAlertFired({
+                    description: t(
+                        "views:components.profile.notifications.updateProfileInfo.success.description"
+                    ),
+                    level: AlertLevels.SUCCESS,
+                    message: t(
+                        "views:components.profile.notifications.updateProfileInfo.success.message"
+                    )
+                });
+
+                // Re-fetch the profile information
+                dispatch(getProfileInformation(true));
+                setShowEditModal(false);
+            }
+        });
+    };
+
+    /**
+     * This is called when the save button is clicked
+     * @param event
```

Comment:
Add param meter comments and leave a space between comment and parameter comment.